consensus/leader.go

```diff
@@ -183,7 +204,11 @@ func (consensus *Consensus) onPrepare(recvMsg *FBFTMessage) {
 	//// Write - End
 
 	//// Read - Start
-	if consensus.decider.IsQuorumAchieved(quorum.Prepare) {
+	quorumFromInitialSignature := hasMultiBlsKeys && isFirstReceivedSignature && quorumPreExisting
+	quorumPostNewSignatures := consensus.decider.IsQuorumAchieved(quorum.Prepare)
+	quorumFromNewSignatures := !quorumPreExisting && quorumPostNewSignatures
+
+	if quorumFromInitialSignature || quorumFromNewSignatures {
```

Comment:
old code was fine, as we just need `consensus.didReachPrepareQuorum()` to hit once